Add QA columns to archived item list

SuaYoo opened the pull request:
- Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1672
- Fixes https://github.com/webrecorder/browsertrix-cloud/issues/1679

<!-- Fixes #issue_number -->

### Changes
- Shows QA review status, latest QA state, and QA run count in archived item list. "Created by" column replaced for space and moved to "Date Created" tooltip
- Enables sort archived items by QA columns
- Adds tooltip to all columns except name
- Hides non-applicable columns in when archived items are filtered to uploads only
- Fixes issue where tooltips weren't activated on hover when using `btrix-table`
- Minor refactor to reuse status labels/colors, renames page review -> page approval to clarify

### Manual testing

1. Log in as any role
2. Go to "Archived Items". Verify "QA Rating" and "Analysis Runs" columns are shown
3. Hover over "QA Rating" and "Analysis Runs" values. Verify tooltip is shown with rating scale and latest analysis run state, respectively
4. Hover over created date. Verify "created by" name is shown
5. Click archived item. Verify navigation to detail page as expected

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Archived Items - All  | <img width="1305" alt="Screenshot 2024-04-22 at 10 36 25 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/4b41c919-3b86-43d7-bb7b-0de6eaff6aab"> |
| Archived Items - All | <img width="321" alt="Screenshot 2024-04-17 at 3 03 18 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/7b8e07fa-72d0-476f-9eaf-3934d0442a99"> |




<!-- ### Follow-ups -->

SuaYoo committed:
update icon

SuaYoo committed:
show conditionally based on item type

SuaYoo committed:
move review status to helpers

SuaYoo committed:
create component for review status

SuaYoo committed:
add tooltips

SuaYoo committed:
adjust column widths

SuaYoo committed:
sort by qa

SuaYoo committed:
update analysis run state

SuaYoo committed:
add click to rest of tooltips

ikreymer commented:
Looks good! I'm not seeing the 'latest QA state' anywhere, but otherwise looks good. Is there another column for it?

Shrinks99 commented:
<img width="680" alt="Screenshot 2024-04-18 at 12 21 18 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/5672810/7af57668-d03f-4413-b6a4-b59b074f1f22">

Would want to continue to use "none" in grey instead of "0" or perhaps "0" in grey?  Good to further distinguish the states between nothing and something existing?  We've also used "---" for this state before

ikreymer commented:
> Would want to continue to use "none" in grey instead of "0" or perhaps "0" in grey? Good to further distinguish the states between nothing and something existing?

I think this makes sense - the None is for no rating while 0 is for QA runs. I guess could say 'No Rating' instead too

ikreymer commented:
PR #1694 adds an `activeQAStats` field to the response, to make it possible to show a progress bar/spinner.
The stats are in the usual found/done format:
```
activeQAState: "running"
activeQAStats: {found: 1, done: 3, size: 526662}
```

ikreymer committed:
fix type warnings

ikreymer commented:
Perhaps should merge this and add remaining tweaks / running progress in separate PR

ikreymer commented:
Could try something like this with microscope icon + color coding, similar to gear icon, for initial pass:
<img width="1351" alt="Screenshot 2024-04-20 at 11 53 01 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/1015759/a60c1991-9ac5-4c16-b8e6-381b155a52bd">

ikreymer commented:
Added the spinner + sorting by 'Last Analysis Started', which might be more intuitive - though perhaps should show the date somewhere?

<img width="1301" alt="Screenshot 2024-04-21 at 1 50 41 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/1015759/9d4b35d7-b589-4e94-8868-da42ec52497f">

ikreymer committed:
add progress ring + sorting by 'last qa run started' (dependent on #1712 to be merged in)

tw4l committed:
Update frontend to match backend changes in #1712

- Rename QA sort fields to `lastQAState` and `lastQAStarted`
- Include active/current QA in last, to be consistent with workflows
- Check if current QA based on `activeQAStats`, since we now removed
`activeQAState` as redundant

tw4l commented:
The progress spinner seems a bit oversized compared to the microscope icon, I would maybe reduce it in size so it's more visually consistent if it stays legible that small? Otherwise I think this is looking very good.

tw4l commented on frontend/src/features/archived-items/archived-item-list.ts:
```diff
@@ -123,38 +177,127 @@ export class ArchivedItemListItem extends TailwindElement {
+                          ? html`
+                              <sl-progress-ring
+                                value="${activeProgress}"
+                                style="color: ${qaStatus.cssColor}; --size: 20px; --track-width: 1px; --indicator-width: 2px;"
```
the microscope icons seem to resolve to about 14px, so maybe that (or something just slightly larger) would be a better size for this indicator?

SuaYoo committed:
adjust styles

SuaYoo committed:
move qa status

SuaYoo committed:
swap column order

SuaYoo committed:
update tooltips

SuaYoo committed:
fix nan progress

SuaYoo commented:
Updated columns to the following:
<img width="1305" alt="Screenshot 2024-04-22 at 10 36 25 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/5fe6d9c8-d62c-455e-8022-f05f6eee3d65">

SuaYoo commented:
> Would want to continue to use "none" in grey instead of "0" or perhaps "0" in grey? Good to further distinguish the states between nothing and something existing?

Check out the latest--tried out the slash icon to denote both nothing and and n/a, tooltip updates accordingly--thoughts?

SuaYoo committed:
consistent state -> status

SuaYoo committed:
update types

SuaYoo committed:
show last run on hover

SuaYoo commented:
Updated QA run tooltip to show last run date:
<img width="368" alt="Screenshot 2024-04-22 at 12 33 56 PM" src="https://github.com/webrecorder/browsertrix/assets/4672952/aeaff887-98b7-451d-882a-af9938b887b5">

cc @tw4l @ikreymer
